app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt

```diff
@@ -114,4 +126,75 @@ open class TripsViewModel(
         }
         .addOnFailureListener { exception -> onFailure(exception) }
   }
+
+  open fun addActivityToTrip(activity: Activity) {
+    if (selectedTrip.value != null) {
+      val trip = selectedTrip.value!!
+      val updatedTrip = trip.copy(activities = trip.activities + activity)
+      updateTrip(
+          updatedTrip,
+          onSuccess = {
+            /*
+                This is a trick to force a recompose, because the reference wouldn't
+                change and update the UI.
+            */
+            selectTrip(Trip())
+            selectTrip(updatedTrip)
+          })
+    }
+  }
+
+  // ****************************************************************************************************
+  // AI assistant
+  // ****************************************************************************************************
+  private val generativeModel =
+      GenerativeModel(
+          modelName = "gemini-1.5-flash",
+          apiKey = BuildConfig.GEMINI_API_KEY,
+          generationConfig =
+              generationConfig {
+                responseMimeType = "application/json"
+                responseSchema =
+                    Schema(
+                        name = "activities",
+                        description = "List of activities",
+                        type = FunctionType.ARRAY,
+                        items =
+                            Schema(
+                                name = "activity",
+                                description = "An activity",
+                                type = FunctionType.OBJECT,
+                                properties =
+                                    mapOf(
+                                        "title" to
+                                            Schema(
+                                                name = "title",
```

Comment:
Consider extracting the schema logic into a separate reusable function or component.

Reply:
We will not reuse the schema with another generative model (maybe we will have models with other schemas, but not this one). So I think we should leave it as is for better readability.